src/NuGet.Core/NuGet.Configuration/PackageSourceMapping/PackageSourceMappingProvider.cs

```diff
@@ -58,5 +58,39 @@ public void AddOrUpdatePackageSourceMappingSourceItem(PackageSourceMappingSource
 
             _settings.SaveToDisk();
         }
+
+        public void SavePackageSourceMappings(IReadOnlyList<PackageSourceMappingSourceItem> packageSourceMappingsSourceItems)
+        {
+            if (packageSourceMappingsSourceItems == null)
+            {
+                throw new ArgumentNullException(nameof(packageSourceMappingsSourceItems));
+            }
+
+            IReadOnlyList<PackageSourceMappingSourceItem> existingSettingsLookup = GetPackageSourceMappingItems();
+            if (existingSettingsLookup == null)
+            {
+                return;
+            }
+            // Remove all old mappings not in new mappings
+            List<PackageSourceMappingSourceItem> removeMappings = new List<PackageSourceMappingSourceItem>();
+            foreach (PackageSourceMappingSourceItem sourceItem in existingSettingsLookup)
+            {
+                if (!packageSourceMappingsSourceItems.Contains(sourceItem))
+                {
+                    removeMappings.Add(sourceItem);
+                }
+            }
+
+            if (removeMappings != null && removeMappings.Count > 0)
```

Comment:
```suggestion
            if (removeMappings.Count > 0)
```
You already initialized the list at line 75 so it's not going to be null.

Reply:
This one is not resolved, so I reverted back to unresolved state.

Reply:
Oops sorry. I clicked the commit suggestion button, so I thought it was resolved. I guess the commit suggestion button does not actually work? I changed it manually